superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterSets.tsx

```diff
@@ -0,0 +1,198 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Input, Select } from 'src/common/components';
+import Button from 'src/components/Button';
+import React, { ChangeEvent, useState } from 'react';
+import { styled, t, tn } from '@superset-ui/core';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  DataMaskState,
+  DataMaskUnitWithId,
+  MaskWithId,
+} from 'src/dataMask/types';
+import { setFilterSetsConfiguration } from 'src/dashboard/actions/nativeFilters';
+import { FiltersSet, FilterSets } from 'src/dashboard/reducers/types';
+import { generateFiltersSetId } from './utils';
+import { Filter } from '../types';
+
+const FilterSet = styled.div`
+  display: grid;
+  align-items: center;
+  justify-content: center;
+  grid-template-columns: 1fr;
+  grid-gap: 10px;
+  padding-top: 10px;
+`;
+
+const StyledTitle = styled.h4`
+  width: 100%;
+  font-size: ${({ theme }) => theme.typography.sizes.s}px;
+  color: ${({ theme }) => theme.colors.grayscale.dark1};
+  margin: 0;
+  overflow-wrap: break-word;
+
+  & > .ant-select {
+    width: 100%;
+  }
+`;
+
+const ActionButtons = styled.div`
+  display: grid;
+  flex-direction: row;
+  grid-template-columns: 1fr 1fr;
+  ${({ theme }) =>
+    `padding: 0 ${theme.gridUnit * 2}px ${theme.gridUnit * 2}px`};
+  border-bottom: 1px solid ${({ theme }) => theme.colors.grayscale.light2};
+
+  .btn {
+    flex: 1;
+  }
+`;
+
+const Sets = styled(ActionButtons)`
+  grid-template-columns: 1fr;
+`;
+
+type FilterSetsProps = {
+  dataMaskState: DataMaskUnitWithId;
+  onFilterSelectionChange: (
+    filter: Pick<Filter, 'id'> & Partial<Filter>,
+    dataMask: Partial<DataMaskState>,
+  ) => void;
+};
+
+const FilterSets: React.FC<FilterSetsProps> = ({
+  onFilterSelectionChange,
+  dataMaskState,
+}) => {
+  const dispatch = useDispatch();
+  const [filtersSetName, setFiltersSetName] = useState('');
+  const filterSets = useSelector<any, FilterSets>(
+    state => state.nativeFilters.filterSets ?? {},
+  );
+  const filterSetsConfigs = useSelector<any, FiltersSet[]>(
+    state => state.dashboardInfo?.metadata?.filter_sets_configuration || [],
+  );
+  const [selectedFiltersSetId, setSelectedFiltersSetId] = useState<
+    string | null
+  >(null);
+
+  const takeFiltersSet = (value: string) => {
```

Comment:
While we're moving these, it would be nice to rename these to comply with the "Filter set" name instead of "Filters set".  In this case, this could be `takeFilterSet`. Also, "id" might be a more descriptive variable name than "value" here.

Reply:
this function will be moved and renamed in the next PR